plane/src/drone/docker/mod.rs

```diff
@@ -166,6 +168,112 @@ impl PlaneDocker {
+            "cpu_stats.system_cpu_usage".into(),
+        ));
+    };
+    let container_cpu_used = stats.cpu_stats.cpu_usage.total_usage;
+    let prev_sys_cpu_ns_load = prev_sys_cpu_ns.load(Ordering::SeqCst);
+    let prev_container_cpu_ns_load = prev_container_cpu_ns.load(Ordering::SeqCst);
+
+    if container_cpu_used < prev_container_cpu_ns_load {
+        return Err(MetricsConversionError::ContainerCpuLessThanCurrent {
+            current: container_cpu_used,
+            prev: prev_container_cpu_ns_load,
+        });
+    }
+    if total_system_cpu_used < prev_sys_cpu_ns_load {
+        return Err(MetricsConversionError::SysCpuLessThanCurrent {
+            current: total_system_cpu_used,
+            prev: prev_sys_cpu_ns_load,
+        });
+    }
+
+    let container_cpu_used_delta =
+        container_cpu_used - prev_container_cpu_ns.load(Ordering::SeqCst);
+
+    let system_cpu_used_delta = total_system_cpu_used - prev_sys_cpu_ns_load;
+
+    prev_container_cpu_ns.store(container_cpu_used, Ordering::SeqCst);
+    prev_sys_cpu_ns.store(total_system_cpu_used, Ordering::SeqCst);
+
+    // NOTE: a BIG limitation here is that docker does not report swap info!
+    // This may be important for scheduling decisions!
+
+    let (mem_total, mem_active, mem_inactive, mem_unevictable, mem_available) = match mem_stats {
+        bollard::container::MemoryStatsStats::V1(v1_stats) => {
+            let active_mem = v1_stats.active_anon + v1_stats.active_file;
+            let total_mem = v1_stats.total_rss + v1_stats.total_cache;
+            let unevictable_mem = v1_stats.total_unevictable;
+            let inactive_mem = v1_stats.inactive_anon + v1_stats.inactive_file;
+            let mem_available = mem_limit - active_mem - unevictable_mem;
+            (
+                total_mem,
+                active_mem,
+                inactive_mem,
+                unevictable_mem,
+                mem_available,
+            )
+        }
+        bollard::container::MemoryStatsStats::V2(v2_stats) => {
+            let active_mem = v2_stats.active_anon + v2_stats.active_file;
+            let kernel = v2_stats.kernel_stack + v2_stats.sock + v2_stats.slab;
+            let total_mem = v2_stats.file + v2_stats.anon + kernel;
+            let unevictable_mem = v2_stats.unevictable;
+            let inactive_mem = v2_stats.inactive_anon + v2_stats.inactive_file;
+            let mem_available =
+                mem_limit - active_mem - unevictable_mem - v2_stats.slab_unreclaimable;
+            (
+                total_mem,
+                active_mem,
+                inactive_mem,
+                unevictable_mem,
+                mem_available,
+            )
+        }
+    };
+
+    Ok(BackendMetricsMessage {
+        backend_id,
+        mem_total,
+        mem_active,
+        mem_inactive,
+        mem_unevictable,
+        mem_available,
+        cpu_used: container_cpu_used_delta,
+        sys_cpu: system_cpu_used_delta,
```

Comment:
These messages previously had a `mem_used` value along with drone and cluster info. Is there a reason why we left those out? https://github.com/drifting-in-space/plane/blob/plane-0.3.x/core/src/messages/agent.rs#L168

Reply:
sorry, based this pr wrong, discussion here: https://github.com/drifting-in-space/plane/pull/543